cumulus-client/src/main/java/mil/army/usace/hec/cumulus/client/controllers/CumulusDssDataUtil.java

```diff
@@ -0,0 +1,31 @@
+package mil.army.usace.hec.cumulus.client.controllers;
+
+import java.time.format.DateTimeFormatter;
+import java.util.StringJoiner;
+import mil.army.usace.hec.cumulus.client.model.Product;
+import mil.army.usace.hec.cumulus.client.model.Watershed;
+
+public final class CumulusDssDataUtil {
+
+    private CumulusDssDataUtil() {
+        throw new AssertionError("Utility class");
+    }
+
+    /**
+     * Builds Dss path string for given Watershed and Product.
+     * @param watershed - Watershed
+     * @param product - Product
+     * @return DSS path String
+     */
+    public static String buildDssPath(Watershed watershed, Product product) {
+        StringJoiner retVal = new StringJoiner("/");
+        String partA = "SHG";
```

Comment:
if any of these parts are blank or undefined in JSON, what do you get here? Wondering if you need null checks. empty pathname parts are fine too //

Reply:
Added null checks to turn into empty strings if parts are blank in JSON